modules/solid_mechanics/src/materials/ConstitutiveModel.C

```diff
@@ -93,13 +95,19 @@ ConstitutiveModel::applyThermalStrain(unsigned qp,
                                       SymmTensor & strain_increment,
                                       SymmTensor & d_strain_dT)
 {
-  if (_has_temp && _t_step != 0)
+  if (_t_step >= 1)
+    _zeroth_step = false;
+
+  if (_t_step >= 2)
+    _first_step = false;
+
```

Comment:
In that case let's leave this as is with the two booleans:  it's a more transparent replacement for the `_t_step` in the materials for the Bison folks so that I can convince them to merge my bison PR.